File: LeetCode/Binary_Search/labs/704_binary_search/704_binary_search.java
public class BinarySearch {
	public static int findNumber(int nums[], int target) {
		int start = 0;
		int end = nums.length;

		while (start <= end) {
			int mid = start + (end - start) / 2;
			if (nums[mid] == target) return mid;
			else if (nums[mid] > target) end = mid - 1;
			else if (nums[mid] < target) start = mid + 1;
		}
		return -1;
	}

	public static void main(String[] args) {
		int[] nums = {1, 2, 3, 4, 5, 6};
		int target = 3;

		int result = findNumber(nums, target);
		System.out.println(result);
	}
}

File: LeetCode/Binary_Search/labs/704_binary_search/704_binary_search.cpp
#include <iostream>
#include <vector>

class BinarySearch {
	public:
		static int findNumber(std::vector<int> nums, int target) {
			int start = 0;
			int end = nums.size() - 1;
	
			while (start <= end) {
				int mid = start + (end - start) / 2; // more recommended to avoid big numbers

				if (nums[mid] == target) return mid;
				else if (nums[mid] > target) end = mid - 1;
				else if (nums[mid] < target) start = mid + 1;
			}
			return (-1);
		}
};

int main(void) {
	std::vector<int> nums = {-1, 0, 3, 5, 9, 12};
	int target = -5;

	int result = BinarySearch::findNumber(nums, target);
	std::cout << result << std::endl;
	return (0);
}
